### PR TITLE
tests: introduce UI tests for core

### PR DESCRIPTION
**Pull request by tsusanka**
I will add some documentation to this, but I feel we should make this easier to use first. Since we do not want the screen fixtures in git, what we could do is to record the screens in CI simultaneously with the hash checking. Then in case some particular test fails, we would deploy the recorded screens somewhere (as HTML to firmware.corp.sldev.cz?) and provide a link to easily see where the difference occurred. Because working with hashes only is quite cumbersome.

This adds an argument for pytest `test_screen` with four options:

- `record`: make screenshots and save them as fixtures
- `hash`: make hashes of the screenshots instead and store them as fixtures
- `test-hash`: test against the hash fixtures
- `test-record`: test against the screenshosts fixtures

Note:
- This needs more attention: https://github.com/trezor/trezor-firmware/commit/c8c240ab9075ad928a162226c71749f834227778#diff-9f7414700342058eee129030ac8f0d30R226
- On NixOS when recording screens we need to provide --basetemp=[somedir], I am not sure why and I will discuss this with adminkos.

### CODE REVIEW

**Comment by tsusanka** [comment]
Also this introduces a new env variable `TREZOR_DISABLE_ANIMATION` which replaces `TREZOR_DISABLE_FADE` and disables animation. It is required when running UI tests.

File: core/Makefile [hunk 1]
```diff
@@ -79,6 +79,15 @@ test_emu_fido2: ## run fido2 device tests
 test_emu_click: ## run click tests
 	cd tests ; ./run_tests_click_emu.sh $(TESTOPTS)
 
+test_emu_ui:  # run ui integration tests
```
Comment (prusnak) [inline core/Makefile]: Single space and double hash, so the correct line is:
`test_emu_ui: ## run ui integration tests`

File: core/Makefile [hunk 2]
```diff
@@ -79,6 +79,15 @@ test_emu_fido2: ## run fido2 device tests
...
+	cd tests ; ./run_tests_device_emu.sh --test_screen=test-hash -m "not skip_ui" $(TESTOPTS)
+
+test_emu_ui_hash:  # create hashes of ui integration test fixtures
```
Comment (prusnak) [inline core/Makefile]: Same as above

File: core/Makefile [hunk 3]
```diff
@@ -79,6 +79,15 @@ test_emu_fido2: ## run fido2 device tests
...
+	cd tests ; ./run_tests_device_emu.sh --test_screen=hash -m "not skip_ui" $(TESTOPTS)
+
+test_emu_ui_record:  # record a full set of new ui fixtures
```
Comment (prusnak) [inline core/Makefile]: Same as above

File: legacy/firmware/protob/messages-debug.options [hunk 4]
```diff
@@ -15,7 +15,8 @@ DebugLinkMemory.memory                  max_size:1024
...
 
 # unused fields
-DebugLinkState.layout_lines             max_count:0
-DebugLinkState.layout_lines             max_size:1
-DebugLinkLayout.lines                   max_size:1
-DebugLinkLayout.lines                   max_count:0
+DebugLinkState.layout_lines             max_count:10
+DebugLinkState.layout_lines             max_size:30
+DebugLinkLayout.lines                   max_count:10
+DebugLinkLayout.lines                   max_size:30
```
Comment (prusnak) [inline legacy/firmware/protob/messages-debug.options]: Let's have this on single lines, that is like this:

```
DebugLinkState.layout_lines             max_count:10 max_size:30
DebugLinkLayout.lines                   max_count:10 max_size:30
```

**Comment by prusnak** [comment]
I left some comments, otherwise nice!

**Comment by tsusanka** [comment]
I am doing some changes in this, I will update this PR soon.

**Comment by tsusanka** [comment]
Replaced by #784

**Pull request closed without merging**